flips/20220725-cadence-borrowContract.md

```diff
@@ -0,0 +1,75 @@
+# borrowContract function for PublicAccount
+
+| Status        | Proposed    												  |
+:-------------- |:------------------------------------------------------------|
+| **FLIP #**    | [NNN](https://github.com/onflow/flow/pull/NNN)  	 		  |
+| **Author(s)** | Deniz Mert Edincik (deniz@edincik.com)  					  |
+|				| Austin Kline (austin@flowty.io)  							  | 
+| **Sponsor**   | 														      |
+| **Updated**   | 2022-07-25                                    			  |
+
+## Objective
+
+This proposed change will allow dynamic import of contracts from an address, by
+using a well known `borrow` dynamic.
+
+## Motivation
+
+There is a popular request, especially from people coming from EVM space, to
+have a way to import contracts from an address by name ( dynamic import
+functionality )
+
+Cadence lacks with dynamic imports, which leads to mass imports in some 
+contracts ( e.g. alchemy, find ) which also causes a lot of contracts to fail, 
+if one of the important contracts cease to exist. 
+
+## User Benefit
+
+- It makes it possible to use patterns that are common to Solidity developers 
+in Flow. 
+- It allows importing and using Contracts with name collision https://github.com
+/onflow/cadence/issues/1171
+- It will prevent single contract breaking the ecosystem
+
+## Design Proposal
+
+We suggest to add an new method to `PublicAccount` called `borrowContract`.
+
+```cadence 
+fun borrowContract<T: &Any>(name: String): T?
```

Comment:
This seems related to the contracts functionality of an account, which is already grouped into ``PublicAccount.Contracts`. Also, this function should also be available for `AuthAccount`s, i.e. be added to `AuthAccount.Contracts`. Maybe also rename the function to just `borrow`, as the other functions don't contain the noun either (e.g. `get`, `add`, etc.)

Reply:
@turbolent how about putting `borrow<&T>(): &T` method to DeployedContract ? I think I can make T somehow inferred automatically right?